src/parse.y

```diff
@@ -1684,15 +1696,21 @@ for_head:
 					key_var->Error("global variable used in for loop");
 				}
 			else
+				{
+				Unref(key_var);
```

Comment:
`key_var` is null on this branch so this unref is superfluous?

Reply:
Thanks, 4 unnecessary `Unref(nullptr)` removed. 6 memory leak fixes remain in the commit.